packages/kbn-search-index-documents/components/result/result_metadata.ts

```diff
@@ -44,10 +44,36 @@ export const resultMetaData = (result: SearchHit): MetaDataProps => ({
   score: result._score,
 });
 
+const MAPPING_TYPE_ORDER = ['semantic_text', 'dense_vector', 'sparse_vector'];
+const SPECIAL_NAME_FIELDS = ['headings'];
+
+export const reorderFieldsInImportance = (fields: FieldProps[]) => {
+  return fields.sort((a, b) => {
+    const specialA = SPECIAL_NAME_FIELDS.indexOf(a.fieldName);
+    const specialB = SPECIAL_NAME_FIELDS.indexOf(b.fieldName);
+
+    if (specialA !== -1 || specialB !== -1) {
+      if (specialA === -1) return 1;
+      if (specialB === -1) return -1;
+      return specialA - specialB;
+    }
+
+    const typeA = MAPPING_TYPE_ORDER.indexOf(a.fieldType!);
+    const typeB = MAPPING_TYPE_ORDER.indexOf(b.fieldType!);
```

Comment:
are these `!` still needed? looking at the type I think this can't be `undefined`, I know looking at the usage it will always have a value.

Reply:
Yep good spot. I removed the exclusion but forgot to update this code.